cmd/up/stignore.go

```diff
@@ -14,18 +14,85 @@
 package up
 
 import (
+	"bufio"
 	"fmt"
+	"io"
 	"io/ioutil"
+	"os"
+	"path"
 	"path/filepath"
 	"strings"
 
 	initCMD "github.com/okteto/okteto/cmd/init"
 	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/config"
 	"github.com/okteto/okteto/pkg/linguist"
 	"github.com/okteto/okteto/pkg/log"
 	"github.com/okteto/okteto/pkg/model"
 )
 
+func addStignoreSecrets(dev *model.Dev) error {
+	for i, folder := range dev.Sync.Folders {
+		stignorePath := filepath.Join(folder.LocalPath, ".stignore")
+		if !model.FileExists(stignorePath) {
+			continue
+		}
+		infile, err := os.Open(stignorePath)
+		if err != nil {
+			return err
+		}
+		defer infile.Close()
+		reader := bufio.NewReader(infile)
+
+		stignoreName := fmt.Sprintf("stignore-%d", i+1)
+		transformedStignorePath := filepath.Join(config.GetDeploymentHome(dev.Namespace, dev.Name), stignoreName)
+		outfile, err := os.OpenFile(transformedStignorePath, os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
can we move the transformation to a separate function so it's easier to test and to understand what's happening?